packages/react-charting/src/components/TreeChart/TreeChart.base.tsx

```diff
@@ -0,0 +1,293 @@
+/* eslint-disable dot-notation */
+/* eslint-disable one-var */
+/* eslint-disable @fluentui/max-len */
+/* eslint-disable prefer-arrow-callback */
+/* eslint-disable react/self-closing-comp */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable no-console */
+/* eslint-disable prefer-const */
+import * as React from 'react';
+import { createRef } from 'react';
+import { select, Selection } from 'd3-selection';
+import { hierarchy, tree } from 'd3-hierarchy';
+
+import { ITreeProps, ITreeState, ITreeDataStructure, IDataStructure } from '../../index';
+
+// Create a parent class for common tree components
+class StandardTree {
+  public treeData: IDataStructure;
+  public svg: Selection<SVGSVGElement | null, unknown, null, undefined>;
+  constructor(treeData: IDataStructure, svg: Selection<SVGSVGElement | null, unknown, null, undefined>) {
+    this.treeData = treeData;
+    this.svg = svg;
+  }
+
+  public addNodeShapetoSVG(
+    name: string,
+    subname: string,
+    xCoordinate: number,
+    yCoordinate: number,
+    fillColor: string,
+    rectangleWidth: number,
+    rectangleHeight: number,
+  ) {
+    this.svg
+      .append('rect')
+      .attr('width', rectangleWidth)
+      .attr('height', rectangleHeight)
+      .attr('x', xCoordinate)
+      .attr('y', yCoordinate)
+      .attr('padding', '10px')
+      .attr('rx', '1')
+      .style('stroke', fillColor)
+      .style('stroke-width', '2px')
+      .style('fill', 'white');
+
+    this.svg
+      .append('text')
+      .style('fill', 'black')
+      .attr('dy', yCoordinate + rectangleHeight / 2.5)
```

Comment:
what do these numbers represent

Reply:
// Text position y = y + rectHeight/2.5, 2.5 is ratio of depth
 // Text position x = x + rectWidth/3.5, 3.5 is ratio of length
 
 Added these comments for each line